app/src/main/java/com/globant/practice/presentation/presenter/WebClientPresenter.java

```diff
@@ -0,0 +1,77 @@
+package com.globant.practice.presentation.presenter;
+
+import com.globant.practice.presentation.model.WebClientState;
+import com.globant.practice.presentation.view.fragment.WebClientView;
+import javax.inject.Inject;
+
+/**
+ * Manages the possible states of the WebClientFragment using a view model called WebClientState,
+ * load the web page on the WebView and shows o dismiss the progress dialog on the WebClientFragment
+ * Created by jonathan.vargas on 4/05/2017.
+ */
+
+public class WebClientPresenter extends BasePresenter<WebClientView> {
+    private WebClientState webClientState;
+
+    /**
+     * Construct method of the WebClientPresenter and initializes the WebClientState model
+     */
+    @Inject
+    public WebClientPresenter() {
+        webClientState = new WebClientState();
+    }
+
+    /**
+     * Initializes the web page load progress
+     *
+     * @param htmlUrl    address of the web page that will be load
+     * @param detailType indicates if the web page are a profile or a repository
+     */
+    public void loadWebPage(String htmlUrl, String detailType) {
+        if (!webClientState.isLoading()) {
+            webClientState.setHtmlUrl(htmlUrl);
+            webClientState.setDetailType(detailType);
+            webClientState.setLoading(true);
+            webClientState.setErrorShowing(false);
+            if (isViewAttached()) {
```

Comment:
I think this should be out of the loading if condition, since the user can change the device orientation, the ui is recreated and the presenter will not render nothing. Let me know if I'm right or wrong